lib/SecretsCrypto/secrets.c

```diff
@@ -0,0 +1,1000 @@
+				      currFilter->field,
+				      g_variant_new("s", currFilter->value));
+		currFilter = currFilter->next;
+	}
+
+	result = g_variant_new(
+			    "(@(ss)@ay@a{sv})",
+			    Sailfish_Secrets_variantFromSecretIdentifier(secret->identifier),
+			    g_variant_builder_end(dataBuilder),
+			    g_variant_builder_end(filterDataBuilder));
+
+	g_variant_builder_unref(dataBuilder);
+	g_variant_builder_unref(filterDataBuilder);
+
+	return result;
+}
+
+struct Sailfish_Secrets_Secret *
+Sailfish_Secrets_secretFromVariant(GVariant *variant)
+{
+	struct Sailfish_Secrets_Secret *secret = NULL;
+	GVariant *identVariant = NULL;
+	char *secretName = NULL;
+	char *collectionName = NULL;
+	GVariantIter *dataIter;
+	unsigned char datum = 0;
+	unsigned char *data =
+	    (unsigned char *)malloc(2048 * sizeof(unsigned char));
+	size_t dataSize = 0;
+	size_t allocatedSize = 2048 * sizeof(unsigned char);
+	GVariantIter *filterIter;
+	char *field = NULL;
+	GVariant *value = NULL;
+
+	g_variant_get(variant, "(@(ss)aya{sv})",
+		      &identVariant, &dataIter, &filterIter);
+
+	memset(data, 0, 2048 * sizeof(unsigned char));
+	while (g_variant_iter_next(dataIter, "y", &datum)) {
+		dataSize += sizeof(unsigned char);
+		if (allocatedSize < dataSize) {
+			unsigned char *newData = NULL;
+			unsigned char *oldData = data;
+			size_t newAllocSize = allocatedSize * 2;
+			newData = (unsigned char *)malloc(newAllocSize);
+			memset(newData, 0, newAllocSize);
+			memcpy(newData, data, allocatedSize);
+			data = newData;
+			free(oldData);
+		}
+		data[dataSize - 1] = datum;
+	}
+
+	secret = Sailfish_Secrets_Secret_new(data, dataSize);
+	free(data);
+	data = NULL;
+
+	g_variant_get(identVariant, "(ss)", &secretName, &collectionName);
+	Sailfish_Secrets_Secret_setIdentifier(secret, secretName,
+					      collectionName);
+
+	while (g_variant_iter_next(filterIter, "{sv}", &field, &value)) {
+		Sailfish_Secrets_Secret_addFilter(secret, field,
+						  g_variant_get_string(value,
+								       NULL));
+		g_variant_unref(value);
+		g_free(field);
+	}
+
+	g_variant_iter_free(dataIter);
+	g_variant_iter_free(filterIter);
+	g_variant_unref(identVariant);
+	g_free(collectionName);
+	g_free(secretName);
+
+	return secret;
+}
+
+/******************************* Internal Secrets Manager *******************/
+
+void Sailfish_Secrets_createCollectionReady(
+		GObject *source_object,
+		GAsyncResult *res,
+		gpointer user_data)
+{
+	struct Sailfish_Secrets_Result *result = NULL;
+	struct Sailfish_Secrets_Callback_Data *cbd =
+	    (struct Sailfish_Secrets_Callback_Data *)user_data;
+	GVariant *daemon_result =
+	    g_dbus_proxy_call_finish(daemon_connection.secretsProxy, res, NULL);
+	(void)source_object;
+	if (daemon_result) {
+		int resultCode = 0;
+		int errorCode = 0;
+		gchar *errorMessage = NULL;
+		g_variant_get(
+		    daemon_result, "((iis))",
+		    &resultCode,
+		    &errorCode,
+		    &errorMessage);
+		result = Sailfish_Secrets_Result_new(
```

Comment:
`result = Sailfish_Secrets_Result_new(resultCode, errorCode,
                                                              errorMessage);`
Better to align to `(` I suppose

Reply:
In many cases it's not possible to do so, while retaining the 80 column limit, due to kernel style's lengthy tab spacing.  Note that I ran lident over the source .c files, and made only minor adjustments where the output had very suboptimal readability.